genie-web/src/main/java/com/netflix/genie/web/configs/GenieNotificationsAutoConfiguration.java

```diff
@@ -53,4 +55,41 @@ public JobStateChangeObserverMulticaster jobStateChangeObserverMulticaster(
         return multicaster;
     }
 
+    /**
+     * Create a {@link JobNotificationMetricPublisher} which publishes metrics related to to job state changes
+     * notifications.
+     *
+     * @param registry the metrics registry
+     * @return a {@link JobNotificationMetricPublisher}
+     */
+    @Bean
+    @ConditionalOnMissingBean(JobNotificationMetricPublisher.class)
+    public JobNotificationMetricPublisher jobNotificationMetricPublisher(
+        final MeterRegistry registry
+    ) {
+        return new JobNotificationMetricPublisher(registry);
+    }
+
+    /**
+     * Create a {@link JobNotificationSNSPublisher} if unless one exists in the context already.
+     *
+     * @param snsClient  Amazon SNS client
+     * @param properties configuration properties
+     * @param registry   the metrics registry
+     * @return a {@link JobNotificationSNSPublisher}
+     */
+    @Bean
+    @ConditionalOnProperty(value = JobNotificationsSNSProperties.ENABLED_PROPERTY, havingValue = "true")
+    @ConditionalOnMissingBean(JobNotificationSNSPublisher.class)
```

Comment:
We are loading the autoconfiguration for SNS so kinda assuming and SNS client is there. 
We make similar assumptions for many other components.

So rather than letting users pull out this component, i'm providing a convenient way to turn it off (it is off by default)

Reply:
Just saying that if they turn it on there's still a chance it could break if for some reason the SNS client wasn't created (removed classes, disabled configuration, etc) like how we protect some other AWS stuff like [this](https://github.com/Netflix/genie/blob/master/genie-common-internal/src/main/java/com/netflix/genie/common/internal/configs/AwsAutoConfiguration.java#L70) even though the auto configs are on the classpath

Reply:
I haven't changed this bit, how do you think i should protect?

Reply:
this is probably fine `ConditionalOnBean(AmazonSNS.class)`